src/state/feed-feedback.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react'
+import {AppState, AppStateStatus} from 'react-native'
+import debounce from 'lodash.debounce'
+
+import {PROD_DEFAULT_FEED} from '#/lib/constants'
+import {FeedDescriptor, isFeedPostSlice} from '#/state/queries/post-feed'
+
+// TODO replace with atproto api
+interface Interaction {
+  uri: string
+  event: string
+  feedContext: string
+}
+
+type StateContext = {
+  enabled: boolean
+  onItemSeen: (item: any) => void
+  sendInteraction: (interaction: Interaction) => void
+}
+
+const stateContext = React.createContext<StateContext>({
+  enabled: false,
+  onItemSeen: (_item: any) => {},
+  sendInteraction: (_interaction: Interaction) => {},
+})
+
+export function Provider({
+  feed,
+  children,
+}: React.PropsWithChildren<{feed: FeedDescriptor}>) {
+  const enabled = isDiscoverFeed(feed)
+  const queue = React.useRef<Set<string>>(new Set())
+
+  const sendToFeed = React.useRef(
+    debounce(
```

Comment:
Nit: this will create a new debounced function on every render (and discard it). There two nicer ways are:

- `useRef()` and then immediately `if (!ref.current) { ref.current = ... }`
- Or just `useState` with unused setter and lazy initializer